src/build/index.ts

```diff
@@ -381,6 +384,24 @@ export default class StreamsBuild {
         }
     }
 
+    private static async checkIfDirty(filePath): Promise<void> {
+        let isFileDirty = false;
+        let dirtyFile = null;
+        const fileName = filePath.substring(filePath.lastIndexOf("/") + 1);
+        for (let i = 0; i < workspace.textDocuments.length; i += 1) {
+            if (workspace.textDocuments[i].fileName === filePath && workspace.textDocuments[i].isDirty) {
+                isFileDirty = true;
+                dirtyFile = workspace.textDocuments[i];
+            }
+        }
+        if (isFileDirty) {
+            const selection = await window.showWarningMessage(`There are unsaved changes in${fileName}(or makefile). Do you want to continue with the unsaved changes or save the file and continue with the build?`, ...['yes', 'no']);
+            if (selection === 'yes') {
+                dirtyFile.save();
+            }
```

Comment:
```suggestion
            const selection = await window.showWarningMessage(
                `There are unsaved changes in ${fileName}. Do you want to save the file before continuing with the build?`,
                ...['Yes', 'No']
            );
            if (selection === 'Yes') {
                await dirtyFile.save();
            }
```
Let's clean up the message a little and make sure to wait for the file to save (using `await`).